Sources/Controllers/MessagesViewController.swift

```diff
@@ -152,6 +135,138 @@ open class MessagesViewController: UIViewController {
             NSLayoutConstraint.activate([top, bottom, trailing, leading])
         }
     }
+
+    // MARK: - UICollectionViewDataSource
+
+    open func numberOfSections(in collectionView: UICollectionView) -> Int {
+        guard let collectionView = collectionView as? MessagesCollectionView else {
+            fatalError(MessageKitError.notMessagesCollectionView)
```

Comment:
TBO, we have much these `fatalError`, maybe we can have a better friendly way to do this (maybe `assert` or just use some default values)? other than just stop execution in `Debug` or `Release`.

Reply:
Yeah, I think the `fatalError`s here are valid. With good documentation in the superclasses this should not be an issue. And if it is; its a valid reason to crash there as there is no way to recover.

Changing these to `assert`s could be an option to prevent crashes in release builds, but I am not sure if that’s the best approach. How does it work with APIs like `UITableView` and `UICollectionView`?

Reply:
Thanks for the feedback @BasThomas! 

> How does it work with APIs like UITableView and UICollectionView?

This is not standard `UITableView` or `UICollectionView` behavior. Instead of extending the UIKit protocols, we conform to them internally and then call our own MessageKit protocols inside of them.  This is to decouple the `MessagesDataSource` from `UICollectionViewDataSource` so the project can be used with things like IGListKit. So we're forced to provide a valid implementation or it is a `fatalError` 😕

Reply:
Seems like a good usage of `fatalError` then. :)

Reply:
test

Reply:
test

Reply:
test

Reply:
test

Reply:
test

Reply:
test

Reply:
Test

Reply:
Test